addons/l10n_in_check_gst_status/models/account_move.py

```diff
@@ -0,0 +1,55 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    l10n_in_partner_gstin_status = fields.Selection([
+            ("active", "Active"),
+            ("cancelled", "Cancelled"),
+            ("invalid", "Invalid"),
+            ("not_checked", "Not Checked"),
+        ],
+        string="GST Status",
+        default="not_checked",
+        compute="_compute_l10n_in_partner_gstin_status_and_date",
+        store=True,
+        help='''The GSTN status color indicates how recently it was checked when it is 'Active'.\n
+                - Green means it was checked within the last 30 days.
+                - Orange means it was checked between 31 and 60 days.
+                - Red means it was checked more than 60 days ago.'''
+    )
+    l10n_in_gstin_verified_date = fields.Date(compute="_compute_l10n_in_partner_gstin_status_and_date", store=True)
+    l10n_in_gstin_status_color = fields.Selection([
+            ("green", "Green"),
+            ("yellow", "Yellow"),
+            ("red", "Red"),
+        ],
+        default='red',
+        store=True,
+        compute="_compute_l10n_in_gstin_status_color",
+    )
+
+    @api.depends('partner_id', 'partner_id.l10n_in_gstin_verified_status', 'partner_id.l10n_in_gstin_verified_date')
+    def _compute_l10n_in_partner_gstin_status_and_date(self):
+        for move in self:
+            if move.payment_state not in ['in_payment', 'paid', 'reversed'] and move.country_code == 'IN' and move.partner_id.vat and move.partner_id.country_id.code == 'IN':
+                move.l10n_in_partner_gstin_status = move.partner_id.l10n_in_gstin_verified_status
+                move.l10n_in_gstin_verified_date = move.partner_id.l10n_in_gstin_verified_date
+
+    @api.depends('partner_id', 'partner_id.l10n_in_gstin_verified_status', 'partner_id.l10n_in_gstin_verified_date')
+    def _compute_l10n_in_gstin_status_color(self):
```

Comment:
I still find this color thing a bit tricky
- Do we need a store compute?
- Do we really need to compute it after the invoice is Paid?
- What if I have not checked the GSTIN number? (why the color should be red? may be replace it with grey?)

We need to verify it with the PO